In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_excel('data.xlsx', sheet_name='Data')

In [3]:
df = df.drop(['KV', 'thunhap', 'chitieu', 'songuoi'],1)

In [4]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [5]:
y = df.pop('Chi tiêu')

In [6]:
df['Khu vực'] = df['Khu vực'].replace('Nông thôn', 0)
df['Khu vực'] = df['Khu vực'].replace('Thành thị', 1)
df['Khu vực'] = df['Khu vực'].astype(int)
df['Khu vực'].head()

0    1
1    1
2    1
3    1
4    1
Name: Khu vực, dtype: int32

In [7]:
from statsmodels.stats.outliers_influence import variance_inflation_factor

df = df.select_dtypes(exclude = 'object')
# VIF dataframe
vif_data = pd.DataFrame()
vif_data["feature"] = df.columns
  
# calculating VIF for each feature
vif_data["VIF"] = [variance_inflation_factor(df.values, i)
                          for i in range(len(df.columns))]
  
print(vif_data)

       feature       VIF
0      Khu vực  2.392896
1     Số người  3.985062
2     Thu nhập  6.754451
3  Chi ăn uống  9.077163


In [8]:

results = sm.OLS(y, sm.add_constant(df)).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:               Chi tiêu   R-squared:                       0.937
Model:                            OLS   Adj. R-squared:                  0.937
Method:                 Least Squares   F-statistic:                     1548.
Date:                Fri, 21 May 2021   Prob (F-statistic):          7.32e-248
Time:                        14:54:22   Log-Likelihood:                -1850.1
No. Observations:                 420   AIC:                             3710.
Df Residuals:                     415   BIC:                             3730.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const           7.4122      2.841      2.609      

</br> 

Kết luận:

- Biến `Khu vực` không có ý nghĩa đối với mô hình về mặt thống kê 
- Biến `Số người` có ý nghĩa đối với mô hình về mặt thống kê (với mức ý nghĩa (***) hay p-value = 0.000)
- Biến `Thu nhập` có ý nghĩa đối với mô hình về mặt thống kê (với mức ý nghĩa (***) hay p-value = 0.000)
- Biến `Chi ăn uống` có ý nghĩa đối với mô hình về mặt thống kê (với mức ý nghĩa (***) hay p-value = 0.000)

- Mô hình có thể giải thích được 93.7% sự thay đổi của biến `charges`
- Mô hình tương đối tốt (p-value = 7.32e-248)

- Phương trình hồi quy:
Chi tiêu $=$ 7.4122 + -5.3675 $\times$ Số người + 0.1913 $\times$ Thu nhập +  1.1856 $\times$ Chi ăn uống